src/box/lua/net_box.lua

```diff
@@ -830,7 +830,9 @@ end
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
     precheck_request(self, opts)
-    return (pcall(self._request_no_precheck, self, M_PING, opts, nil, self._stream_id))
+    local status, res = pcall(self._request_no_precheck, self, M_PING,
+                              opts, nil, self._stream_id)
+    return status and res or false
```

Comment:
Can't you just `return status` here? Then you wouldn't need to patch `net_box.c`.